Добавляет рецепт «Как делать пулреквест» (#4721)
doka-guide/content: recipes/github-new-pull-request/index.md

---
title: "Как сделать пулреквест на GitHub"

> description: "Создайте свой первый запрос на изменение GitHub pull request."

Reviewer: ```suggestion
description: "Создайте свой первый запрос на изменение GitHub пулреквеста."
```

Reviewer: И можно добавить в список ключевых слов «pull request».

authors:
  - igsekor
related:
  - tools/version-control
  - tools/git-cli
  - tools/github-actions
tags:
  - article
---

## Задача

> Сделать пулреквест в существующий репозиторий на GitHub

Reviewer 2: ```suggestion
Сделать пулреквест в существующий репозиторий на GitHub.
```

😁

## Готовое решение

> Суть пулреквеста — создать запрос на изменение в репозиторий. Обычно запрос на изменения сопровождается отзывом (ревью) со стороны других пользователей репозитория, обладающими правами на внесение изменений.

Reviewer 2: ```suggestion
Суть пулреквеста — создать запрос на внесение изменений в репозиторий. Обычно такой запрос сопровождается отзывом (ревью) со стороны других пользователей репозитория, обладающих правами на внесение изменений.
```

Попыталась сделать предложение глаже. Т.к. обычно говорим или "изменение в репозиториИ" или "внесение изменений в репозиториЙ".

> Для того, чтобы сделать пулреквест, необходимо создать отдельную ветку и все правки внести именно в ней. Название ветки можно выбрать произвольным образом, но лучше отразить в нескольких английских словах, написанных через знак переноса (`-`), суть изменений. Например, чтобы внести изменения (пофиксить), можно использовать слово `fix` или `hotfix` (для срочных изменений) в качестве первого слова:

Reviewer: ```suggestion
Для того, чтобы сделать пулреквест, нужно создать отдельную ветку и внести все правки именно в неё. Название ветки можно выбрать произвольным образом, но лучше отразить суть изменений в нескольких английских словах, перечисленных через знак переноса (`-`). Например, чтобы внести изменения (пофиксить), можно использовать слово `fix` или `hotfix` (для срочных изменений) в качестве первого слова:
```

```bash
git checkout -b hotfix
```

Эта команда создаст ветку с именем `hotfix`, и Git переключит репозиторий на новую ветку. Теперь можно вносить изменения с помощью коммитов, например:

```bash
git commit -m 'Вносит правки в описание'
```

Когда все правки внесены, необходимо отправить изменения на GitHub. Для этого нужно выполнить команду:

```bash
git push -u origin hotfix
```

После этого нужно перейти на сайт GitHub и зайти там в репозиторий. Сверху появится сообщение на жёлтом фоне, в котором вам предложат создать пулреквест:

> ![Сообщение с кнопкой на создание пулреквеста](images/compare-and-pull-request.png)

Reviewer: ```suggestion
![Сообщение «hotfix had recent pushes less than a minute ago». После него кнопка «Compare and pull request.»](images/compare-and-pull-request.png)
```

Можно нажать кнопку «Compare & pull request» в этом сообщении или создать пулреквест «вручную» на странице со списком пулреквестов с помощью кнопки «New pull request»:

> ![Кнопка для создания пулреквеста на странице со списком пулреквестов](images/new-pull-request.png)

Reviewer: ```suggestion
![Кнопка «New pull request» для создания пулреквеста на странице со списком пулреквестов. Находится после поля для поиска по проекту с фильтрами.](images/new-pull-request.png)
```

> Далее вас перекинет на страницу с настройками будущего пулреквеста, на которой вы можете выбрать название и описать изменения, выбрать ветку, в которую вы хотите внести изменения (по умолчанию выбирается ветка, из которой вы создавали текущую ветку с изменениями), связать ваш запрос с текущими нуждами, список которых обычно находится в списке ишью, добавить ревьюеров для проведения тщательного анализа изменений и тестирования:

Reviewer: ```suggestion
Далее вас перекинет на страницу с настройками будущего пулреквеста. На ней можете добавить название и описать изменения, выбрать ветку, в которую хотите внести изменения (по умолчанию выбирается ветка, из которой создавали текущую ветку с изменениями) или связать ваш запрос с текущими нуждами. Обычно они находятся в списке ишью. Также можете добавить ревьюеров для проверки изменений и тестирования:
```

![Страница с настройками пулреквеста](images/open-pull-request.png)

Внизу страницы будут перечислены коммиты с изменениями.

> Когда вы будете уверены, что заполнили необходимые поля и внесли необходимые настройки, нажмите кнопку «Create pull request».

Reviewer: ```suggestion
Когда будете уверены, что заполнили необходимые поля и выбрали подходящие настройки, нажмите кнопку «Create pull request».
```
